File: rerun_cpp/src/rerun/archetypes/cylinders3d_ext.cpp
#if 0

#include "cylinders3d.hpp"

namespace rerun::archetypes {
    // <CODEGEN_COPY_TO_HEADER>

    /// Creates a new `Cylinders3D` with the given axis-aligned lengths and radii.
    ///
    /// For multiple cylinders, you should generally follow this with
    /// `Cylinders3D::with_centers()` and one of the rotation methods, in order to move them
    /// apart from each other.
    static Cylinders3D from_lengths_and_radii(
        const Collection<rerun::components::Length>& lengths, const Collection<rerun::components::Radius>& radii) {
        return Cylinders3D().with_lengths(lengths).with_radii(radii);

    // </CODEGEN_COPY_TO_HEADER>
} // namespace rerun::archetypes

#endif

File: crates/store/re_types/src/archetypes/cylinders3d_ext.rs
use super::Cylinders3D;

impl Cylinders3D {
    /// Creates a new [`Cylinders3D`] with the given axis-aligned lengths and radii.
    ///
    /// For multiple cylinders, you should generally follow this with
    /// [`Cylinders3D::with_centers()`] and one of the rotation methods, in order to move them
    /// apart from each other.
    #[inline]
    pub fn from_lengths_and_radii(
        lengths: impl IntoIterator<Item = impl Into<crate::components::Length>>,
        radii: impl IntoIterator<Item = f32>,
    ) -> Self {
        Self::new(lengths, radii)
    }
}
